src/commands/cmd_dispatcher.c

```diff
@@ -0,0 +1,74 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "commands.h"
+#include "cmd_context.h"
+#include <assert.h>
+#include <strings.h>
+
+// Command handler function pointer.
+typedef void(*Command_Handler)(void *args);
+
+// Get command handler.
+static Command_Handler get_command_handler(GRAPH_Commands cmd) {
+	switch(cmd) {
+	case CMD_QUERY:
+		return Graph_Query;
+	case CMD_EXPLAIN:
+		return Graph_Explain;
+	case CMD_PROFILE:
+		return Graph_Profile;
+	default:
+		assert(false);
+	}
+	return NULL;
+}
+
+// Convert from string representation to an enum.
+static GRAPH_Commands determine_command(const char *cmd_name) {
+	if(strcasecmp(cmd_name, "graph.QUERY") == 0) return CMD_QUERY;
+	if(strcasecmp(cmd_name, "graph.EXPLAIN") == 0) return CMD_EXPLAIN;
+	if(strcasecmp(cmd_name, "graph.PROFILE") == 0) return CMD_PROFILE;
+
+	assert(false);
+	return CMD_BULK_UNKNOWN;
+}
+
+int CommandDispatch(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+	CommandCtx *context;
+	if(argc < 3) return RedisModule_WrongArity(ctx);
+
+	const char *command_name = RedisModule_StringPtrLen(argv[0], NULL);
+	const char *graph_name = RedisModule_StringPtrLen(argv[1], NULL);
```

Comment:
This conversion doesn't seem helpful any longer; `GraphContext_Retrieve` can just take a RedisModuleString argument.